test/e2e/kubelet_stats.go

```diff
@@ -600,6 +600,40 @@ func (r *resourceMonitor) GetLatest() (resourceUsagePerNode, error) {
 	return result, utilerrors.NewAggregate(errs)
 }
 
+func (r *resourceMonitor) GetMasterNodeLatest(usagePerNode resourceUsagePerNode) resourceUsagePerNode {
+	result := make(resourceUsagePerNode)
+	var masterUsage resourceUsagePerContainer
+	var nodesUsage []resourceUsagePerContainer
+	for node, usage := range usagePerNode {
+		if strings.HasSuffix(node, "master") {
+			masterUsage = usage
+		} else {
+			nodesUsage = append(nodesUsage, usage)
+		}
+	}
+	nodeAvgUsage := make(resourceUsagePerContainer)
+	for _, nodeUsage := range nodesUsage {
+		for c, usage := range nodeUsage {
+			if _, found := nodeAvgUsage[c]; !found {
+				nodeAvgUsage[c] = &containerResourceUsage{Name: usage.Name}
+			}
+			nodeAvgUsage[c].CPUUsageInCores += usage.CPUUsageInCores
+			nodeAvgUsage[c].MemoryRSSInBytes += usage.MemoryRSSInBytes
+			nodeAvgUsage[c].MemoryWorkingSetInBytes += usage.MemoryWorkingSetInBytes
+			nodeAvgUsage[c].MemoryRSSInBytes += usage.MemoryRSSInBytes
+		}
+	}
+	for c := range nodeAvgUsage {
+		nodeAvgUsage[c].CPUUsageInCores /= float64(len(nodesUsage))
```

Comment:
This is not true per se, as interval may have different lengths. They shouldn't differ too greatly so it's probably OK to report that. It's good to keep it in mind as soon as we're going to take some actions on such data.